addons/binding/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/handler/PentairEasyTouchHandler.java

```diff
@@ -0,0 +1,426 @@
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.pentair.PentairBindingConstants;
+import org.openhab.binding.pentair.internal.PentairPacket;
+import org.openhab.binding.pentair.internal.PentairPacket_HeatSetPoint;
+import org.openhab.binding.pentair.internal.PentairPacket_Status;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PentairEasyTouchHandler} is responsible for implementation of the EasyTouch Controller. It will handle
+ * commands sent to a thing and implements the different channels. It also parses/disposes of the packets seen on the
+ * bus from the controller.
+ *
+ * @author Jeff James - Initial contribution
+ */
+public class PentairEasyTouchHandler extends PentairBaseThingHandler {
+
+    protected Logger logger = LoggerFactory.getLogger(PentairEasyTouchHandler.class);
+
+    /**
+     * current/last status packet recieved, used to compare new packet values to determine if status needs to be updated
+     */
+    protected PentairPacket_Status p29cur = new PentairPacket_Status();
+    /** current/last heat set point packet, used to determine if status in framework should be updated */
+    protected PentairPacket_HeatSetPoint phspcur = new PentairPacket_HeatSetPoint();
+
+    public PentairEasyTouchHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing EasyTouch - Thing ID: {}.", this.getThing().getUID());
+
+        m_id = ((BigDecimal) getConfig().get("id")).intValue();
+
+        // make sure there are no exisitng EasyTouch controllers
+        PentairBaseBridgeHandler bh = (PentairBaseBridgeHandler) this.getBridge().getHandler();
+        List<Thing> things = bh.getThing().getThings();
+
+        for (Thing t : things) {
+            if (t.getUID() == this.getThing().getUID()) {
+                continue;
+            }
+            if (t.getThingTypeUID().equals(EASYTOUCH_THING_TYPE)) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Another EasyTouch controller is already configured.");
+                return;
+            }
+        }
+
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Thing {} disposed.", getThing().getUID());
+        updateStatus(ThingStatus.OFFLINE);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // TODO: expand number of controllable channels (most today are readonly)
+
+        if (command instanceof RefreshType) {
+            logger.debug("EasyTouch received refresh command");
+
+            updateChannel(channelUID.getId(), null);
```

Comment:
Why does it null out the value what is consequence of that what is the reason? Does it trigger a reload?